cmd/kubeadm/app/cmd/join.go

```diff
@@ -419,6 +420,14 @@ func (j *joinData) Run() error {
 		fmt.Printf("[join] Running pre-flight checks before initializing the new control plane instance\n")
 		preflight.RunInitMasterChecks(utilsexec.New(), initCfg, j.ignorePreflightErrors)
 
+		fmt.Println("[join] Pulling control-plane images")
+		containerRuntime, err := utilruntime.NewContainerRuntime(utilsexec.New(), j.cfg.GetCRISocket())
+		if err != nil {
+			return err
+		}
+		if err := preflight.RunPullImagesCheck(utilsexec.New(), initCfg, j.ignorePreflightErrors); err != nil {
```

Comment:
RunPullImagesCheck() already creates a container runtime instance:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks.go#L1042-L1053

Reply:
yop thx @neolit123 i was already fixing it?

Reply:
done